translator/trace/jaeger/protospan_to_jaegerproto_test.go

```diff
@@ -75,38 +74,44 @@ func TestOCProtoToJaegerProto(t *testing.T) {
 			continue
 		}
 
-		// Sort tags to help with comparison, not only for jaeger.Process but also
-		// on each span.
-		sort.Slice(gotJBatch.Process.Tags, func(i, j int) bool {
-			return gotJBatch.Process.Tags[i].Key < gotJBatch.Process.Tags[j].Key
-		})
-		sort.Slice(wantJBatch.Process.Tags, func(i, j int) bool {
-			return wantJBatch.Process.Tags[i].Key < wantJBatch.Process.Tags[j].Key
+		// Sort the got and want jaeger batch values for comparison purposes.
+		sortJaegerProtoBatch(gotJBatch)
+		sortJaegerProtoBatch(wantJBatch)
+
+		if diff := cmp.Diff(gotJBatch, wantJBatch); diff != "" {
+			// Note: Lines with "-" at the beginning refer to values in gotJBatch. Lines with "+" refer to values in
+			// wantJBatch.
+			t.Errorf("OC Proto to Jaeger Proto failed with following difference: \n%s", diff)
+		}
+	}
+}
+
+// Helper method to sort jaeger.Batch for cmp.Diff.
+func sortJaegerProtoBatch(batch *jaeger.Batch) {
+	// First sort the process tags.
+	sort.Slice(batch.Process.Tags, func(i, j int) bool {
+		return batch.Process.Tags[i].Key < batch.Process.Tags[j].Key
+	})
+	// Sort the span tags and the span log fields.
+	for _, jSpan := range batch.Spans {
+
+		// cmp.Diff ends up comparing batch.Spans[].Process where it doesn't handle nil references well.
+		// For reading purposes https://github.com/google/go-cmp/issues/61
+		// To make cmp.Diff bypass the panic, set the Process of the Span to be that of the Batch only for
+		// testing purposes.
+		jSpan.Process = batch.Process
+
+		sort.Slice(jSpan.Tags, func(i, j int) bool {
+			return jSpan.Tags[i].Key < jSpan.Tags[j].Key
 		})
-		var jSpans []*jaeger.Span
-		jSpans = append(jSpans, gotJBatch.Spans...)
-		jSpans = append(jSpans, wantJBatch.Spans...)
-		for _, jSpan := range jSpans {
-			sort.Slice(jSpan.Tags, func(i, j int) bool {
-				return jSpan.Tags[i].Key < jSpan.Tags[j].Key
+		for _, jSpanLog := range jSpan.Logs {
+			sort.Slice(jSpanLog.Fields, func(i, j int) bool {
+				return jSpanLog.Fields[i].Key < jSpanLog.Fields[j].Key
 			})
 		}
-		for _, jSpan := range jSpans {
-			for _, jSpanLog := range jSpan.Logs {
-				sort.Slice(jSpanLog.Fields, func(i, j int) bool {
-					return jSpanLog.Fields[i].Key < jSpanLog.Fields[j].Key
-				})
-			}
-		}
 
-		gjson, _ := json.Marshal(gotJBatch)
```

Comment:
Was removing of json comparison section intentional in this PR?

Reply:
Yes, it was since the test ended up just comparing a normalized string of the json representation of the objects to compare against expected results. cmp.Diff compares objects and their values (even nested objects/values).